scripts/js/footer.js

```diff
@@ -399,45 +399,28 @@ function apiFailure(data) {
   }
 }
 
-// Method to compare two versions. Returns 1 if v2 is smaller, -1 if v1 is
-// smaller, 0 if equal
-// Credits: https://www.geeksforgeeks.org/compare-two-version-numbers/
+// Method to compare two versions.
+// Returns 1 if v2 is smaller, -1 if v1 is smaller, 0 if equal.
 function versionCompare(v1, v2) {
-  // vnum stores each numeric part of version
-  let vnum1 = 0;
-  let vnum2 = 0;
-
-  // Remove possible leading "v" in v1 and v2
-  if (v1[0] === "v") {
-    v1 = v1.substring(1);
-  }
-
-  if (v2[0] === "v") {
-    v2 = v2.substring(1);
+  // Check if both versions are strings
+  if (typeof v1 !== "string" || typeof v2 !== "string") {
+    return 0;
```

Comment:
I don't think we should return 0 here. If they are not type string, we should error as something went wrong. We should not pretend we did the check and they are equal.